tests/vxlan/test_vxlan_ecmp.py

```diff
@@ -621,13 +662,515 @@ def dump_self_info_and_run_ptf(self, tcname, encap_type, expect_encap_success):
+            for intf in selected_intfs:
+                self.setup['duthost'].shell("sudo config interface startup {}".format(intf))
+            logger.info("Shutdown other interfaces.")
+            remaining_interfaces = list(set(all_intfs) - set(selected_intfs))
+            for intf in remaining_interfaces:
+                self.setup['duthost'].shell("sudo config interface shutdown {}".format(intf))
+            downed_bgp_neighbors = get_downed_bgp_neighbors(remaining_interfaces, minigraph_facts)
+            pytest_assert(wait_until(300, 30, 0, bgp_established, self.setup['duthost'], downed_bgp_neighbors), "BGP neighbors didn't come up after all interfaces have been brought up.")
+            self.setup[encap_type]['t2_ports'] = get_corresponding_ports(selected_intfs, minigraph_facts)
+            self.dump_self_info_and_run_ptf("tc12", encap_type, True, packet_count=1000)
+
+            logger.info("Recovery. Bring all up, and verify traffic works.")
+            for intf in all_intfs:
+                self.setup['duthost'].shell("sudo config interface startup {}".format(intf))
+            logger.info("Wait for all bgp is up.")
+            pytest_assert(wait_until(300, 30, 0, bgp_established, self.setup['duthost']), "BGP neighbors didn't come up after all interfaces have been brought up.")
+            logger.info("Verify traffic flows after recovery.")
+            self.setup[encap_type]['t2_ports'] = all_t2_ports
+            self.dump_self_info_and_run_ptf("tc12", encap_type, True, packet_count=1000)
+
+        except Exception:
+            # If anything goes wrong in the try block, atleast bring the intf back up.
+            self.setup[encap_type]['t2_ports'] = all_t2_ports
+            for intf in all_intfs:
+                self.setup['duthost'].shell("sudo config interface startup {}".format(intf))
+            pytest_assert(wait_until(300, 30, 0, bgp_established, self.setup['duthost']), "BGP neighbors didn't come up after all interfaces have been brought up.")
+            raise
+
+    def test_vxlan_remove_add_underlay_default(self, setUp, minigraph_facts, encap_type):
+        '''
+           tc13: remove the underlay default route.
+        '''
+        self.setup = setUp
+
+        logger.info("Find all the underlay default routes' interfaces. This means all T2 interfaces.")
+        all_intfs = list(get_portchannels_to_neighbors(self.setup['duthost'], "T2", minigraph_facts))
+        if not all_intfs:
+            all_intfs = get_ethernet_to_neighbors("T2", minigraph_facts)
+        logger.info("Dumping T2 link info: {}".format(all_intfs))
+        if not all_intfs:
+            raise RuntimeError("no interface found connected to t2 neighbors. pls check the testbed, aborting.")
+
+        try:
+            logger.info("Bring down the T2 interfaces.")
+            for intf in all_intfs:
+                self.setup['duthost'].shell("sudo config interface shutdown {}".format(intf))
+
```

Comment:
```suggestion
time.sleep(10)
```
or `wait_until` with passing `all_intfs` as ignore_list, but only if the issue mentioned above in bgp_established will be fixed.
```suggestion
pytest_assert(wait_until(60, 30, 0, bgp_established, self.setup['duthost'], all_intfs), "BGP neighbors didn't come down after all interfaces have been brought down.")
```